Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine.swift

```diff
@@ -284,13 +284,28 @@ extension ExperienceStateMachine {
             // be looking for target elements on the view that need to be fully loaded first, and this container
             // may be presenting after a pre-step navigation action that is finalizing the loading of the new view.
             DispatchQueue.main.asyncAfter(deadline: .now() + 0.0) {
-                do {
-                    try package.stepDecoratingTraitUpdater(stepIndex.item, nil)
-                    SdkMetrics.renderStart(experience.requestID)
-                    try package.presenter {
-                        try? machine.transition(.renderStep)
+
+                func presentStep(onError: @escaping (Error) -> Void) {
+                    do {
+                        try package.stepDecoratingTraitUpdater(stepIndex.item, nil)
+                        SdkMetrics.renderStart(experience.requestID)
+                        try package.presenter {
+                            try? machine.transition(.renderStep)
+                        }
+                    } catch {
+                        if let error = error as? AppcuesTraitError,
```

Comment:
FYI, there's `catch let error as AppcuesTraitError`. Not sure that's actually better in this circumstance.

You could also `catch let error as AppcuesTraitError were error.retryMilliseconds != nil` but that doesn't actually unwrap the optional.

Mostly just noting the options for my own interest because I can never remember the syntax.

Reply:
yeah I had `catch let error as AppcuesTraitError` but then I still needed the unwrap and a conditional case _then_ another catch block for any other error (maybe never actually used), resulting in multiple calls to `onError`, so I updated to this form to just have the two conditions.